File: jobs/2026-06-16__03-34-54/capacitor_custom_android_echo_pl__FvyubsX/artifacts/user/myapp/android/app/src/main/java/com/example/myapp/EchoPlugin.java
package com.example.myapp;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.JSObject;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Echo")
public class EchoPlugin extends Plugin {

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");
        JSObject result = new JSObject();
        result.put("value", value);
        call.resolve(result);
    }
}

File: jobs/2026-07-03__05-39-37/capacitor_custom_ios_echo_plugin__GUHA73x/artifacts/user/myapp/ios/App/App/EchoPlugin.swift
import Foundation
import Capacitor

/**
 * A custom local Capacitor plugin that echoes back the `value` argument
 * provided by the JavaScript layer.
 *
 * Conforms to `CAPPlugin` (the base class) and `CAPBridgedPlugin` (the modern
 * Capacitor v8 plugin protocol that exposes `identifier`, `jsName`, and
 * `pluginMethods` to the runtime).
 */
@objc(EchoPlugin)
public class EchoPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "EchoPlugin"
    public let jsName = "Echo"

    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": value
        ])
    }
}